lib/sles4sap/sap_deployment_automation_framework/networking.pm

```diff
@@ -0,0 +1,389 @@
+
+        # Attempt to acquire network file lease (update modification time)
+        # This will prevent other tests from spending time checking this file since it is already taken.
+        return if !acquire_network_file_lease(network_lease_file => $lease_file);
+
+    }
+    return $lease_file . '/26';    # Add /26 suffix
+}
+
+=head2 create_lease_file
+
+    create_lease_file(network_space=>'192.168.1.0' [, storage_account=>'SHODAN-storage']);
+
+=over 2
+
+=item B<storage_account>: Storage account containing lease file
+
+=item B<network_space>: Network space to create the lease file for
+
+=back
+
+Creates an uploads new network lease file
+
+=cut
+
+sub create_lease_file {
+    my (%args) = @_;
+    $args{storage_account} //= get_var('SDAF_TFSTATE_STORAGE_ACCOUNT');
+    foreach ('network_space', 'storage_account') {
+        croak "Missing mandatory argument: '$_'" unless $args{$_};
+    }
+    my $network_lease_filename = $args{network_space};
+    my $lease_file = "/tmp/$network_lease_filename";
+    assert_script_run("touch $lease_file", quiet => 1);
+    az_storage_blob_upload(
+        container_name => 'network-spaces',
+        storage_account_name => $args{storage_account},
+        file => $lease_file
+    );
+}
+
+
+=head2 create_new_address_space
+
+    create_new_address_space(deployer_vnet_name=>'SHODAN-vnet' [, timeout=>9001]);
+
+=over 2
+
+=item B<deployer_vnet_name>: Deployer virtual network
+
+=item B<timeout>: Timeout for creating new lease file
+
+=back
+
+Used to assign network space which does not yet have blob file created in 'SDAF_TFSTATE_STORAGE_ACCOUNT' storage account.
+Function generates random address space and checks for an existing lease file in storage account. If file does not exist
+it will be created, otherwise function searches again for network space without existing file.
+Before assigning network space there is a check for an existing peering between this network and deployer vnet.
+This is to avoid assigning network which was created without a lease file.
+
+=cut
+
+sub create_new_address_space {
+    my (%args) = @_;
+    croak 'Missing mandatory argument: deployer_vnet_name' unless $args{deployer_vnet_name};
+    $args{timeout} //= 300;    # Timeout for creating new lease file
+    my $address_space;
+    my $generated_address_space;
+    my $start_time = time();
+
+    record_info('New lease', 'All existing network files in usage: Creating new file');
+    while (!$address_space) {
+        die('Failed to reserve free address space within timeout') if (time() - $start_time > $args{timeout});
+        ($generated_address_space = calculate_net_addr_space()) =~ s/\/26//;
+        # Calculate another network if file for this network is already present
+        sleep 1;    # one sec delay to prevent command spamming
+        next if grep { $_ eq $generated_address_space } @{list_network_lease_files()};
```

Comment:
Doesn't `list_network_lease_files` returns a JSON object? Wondering if there may be problems by treating it as a list, or if the `query => [].name` prevents this.

Reply:
It returns normal perl structure. 
`list_network_lease_files` returns output from `az_storage_blob_list`
and that is : `return decode_json(script_output($az_cmd));`
So it's already decoded.

Reply:
Yes, but `decode_json` returns a hash, not an array. That's my question. Not sure if it can cause problems.

Reply:
No, because `query => [].name` specifies to return an array of names.